src/notation/internal/masternotation.cpp

```diff
@@ -140,6 +140,10 @@ static void clearMeasures(mu::engraving::MasterScore* masterScore)
 {
     TRACEFUNC;
 
+    for (Part* part : masterScore->parts()) {
+        part->removeNonPrimaryInstruments();
```

Comment:
Sorry, one more thing to test: suppose a template already contains part scores (i.e. `Excerpts`) _and_ contains an instrument change. Then I think the instruments corresponding to the instrument changes would also need to be removed from the `parts` of the `Excerpts` scores, not only from the master score. Otherwise, the instrument lists of the `Excerpt` scores parts are not in sync anymore with the master score's parts.

That seems not complicated to realise: just move the `parts` loop inside the loop over `scoreList()` (and replace `masterScore->parts()` with `score->parts()`).